src/core/ext/filters/client_channel/service_config.h

```diff
@@ -96,6 +105,26 @@ class ServiceConfig : public RefCounted<ServiceConfig> {
   static RefCountedPtr<T> MethodConfigTableLookup(
       const SliceHashTable<RefCountedPtr<T>>& table, const grpc_slice& path);
 
+  /// Retrieves the parsed object at index \a index.
+  ServiceConfigParsedObject* GetParsedServiceConfigObject(int index) {
+    GPR_DEBUG_ASSERT(index < registered_parsers_count);
+    return parsed_service_config_objects[index].get();
+  }
+
+  typedef UniquePtr<ServiceConfigParsedObject> (*ServiceConfigParser)(
```

Comment:
Instead of making the parser a simple function, I suggest making it a class.  It will need at least two methods, one for parsing the global parameters (replacing the mechanism currently provided by `ParseGlobalParams()`) and one for parsing per-method parameters (replacing the mechanism currently provided by `CreateMethodConfigTable()`).  Also, this will make it possible for the caller to pass any necessary external state to the parsing object via its ctor.

Reply:
I was thinking of using the parser function to invoke the existing ParseGlobalParams and CreateMethodConfigTable, but making it a class will also work